test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -176,59 +173,66 @@ jruby-config :- jruby-schemas/JRubyConfig
+
+          (testing "valid credentials"
+            (let [auth "{ :basic_auth => { :user => 'foo', :password => 'bar' }}"]
+              (.runScriptlet sc (format "$response = $c.post('%s', 'foo', options: %s)" url auth)))
+            (is (= 200 (.runScriptlet sc "$response.code")))
+            (is (= "hi" (.runScriptlet sc "$response.body"))))
+
+          (testing "invalid credentials"
+            (let [auth "{ :basic_auth => { :user => 'foo', :password => 'baz' }}"]
+              (.runScriptlet sc (format "$response = $c.post('%s', 'foo', options: %s)" url auth)))
+            (is (= 401 (.runScriptlet sc "$response.code")))
+            (is (= "access denied" (.runScriptlet sc "$response.body")))))))))
 
 (deftest http-compressed-requests
   (jetty9/with-test-webserver ring-app-decompressing-gzipped-request port
     (with-scripting-container sc
-      (with-http-client sc port {:use-ssl false}
-        (testing "GZIP compression format"
-          (let [compress "{ :compress => :gzip }"
-                body "howdy"]
-            (.runScriptlet sc (format "$response = $c.post('/', '%s', {}, %s)"
-                                      body
-                                      compress))
-            (is (= "200" (.runScriptlet sc "$response.code")))
-            (is (= body (.runScriptlet sc "$response.body")))))
-
-       (testing "invalid compression format"
-         (let [compress "{ :compress => :bunk }"]
-           (is (= "Unsupported compression specified for request: bunk"
-                  (.runScriptlet
-                   sc
-                   (str "begin;"
-                        "  $response = $c.post('/', 'foo', {}, "
-                        compress
-                        ");"
-                        "  'No error raised from post';"
-                        "rescue ArgumentError => e;"
-                        "  e.message;"
-                        "end"))))))))))
+      (with-http-client sc {}
+        (let [url (str "http://localhost:" port)]
+          (testing "GZIP compression format"
+            (let [compress "{ :compress => :gzip }"
+                  body "howdy"]
+              (.runScriptlet sc (format "$response = $c.post('%s', '%s', options: %s)"
+                                        url
+                                        body
+                                        compress))
+              (is (= 200 (.runScriptlet sc "$response.code")))
+              (is (= body (.runScriptlet sc "$response.body")))))
+
+          (testing "invalid compression format"
+            (let [compress "{ :compress => :bunk }"]
+              (is (= "Unsupported compression specified for request: bunk"
+                     (.runScriptlet
+                      sc
+                      (str "begin;"
+                           "  $response = $c.post('"
+                           url
+                           "', 'foo', options: "
```

Comment:
why put this on three separate lines now?

Reply:
Before `url` wasn't interpolated. This can probably be refactored into a `format` call like the others.